alpa/pipeline_parallel/compile_executable.py

```diff
@@ -245,8 +251,16 @@ def compile_pipeshard_executable_internal(
     debug_compilation_time("shard stages")
 
     # Launch the physical mesh group
-    if virtual_mesh.launched_physical_mesh_group is None:
-        virtual_mesh.get_physical_mesh_group(sliced_virtual_meshes)
+    # if virtual_mesh.launched_physical_mesh_group is None:
+    #     virtual_mesh.get_physical_mesh_group(sliced_virtual_meshes)
+
+    nccl_instantiated = False
+    if 'virtual_meshes' in globals() and virtual_meshes is not None and virtual_mesh.launched_physical_mesh_group is not None:
```

Comment:
please avoid using a global `virtual_meshes`. Instead, please:
1. Always create a virtual mesh group without launching NCCL. Remember we always don't want to touch any physical resources related information in this part.
2. Launch meshes and nccl groups on real resources in the `PipeShardDriverExecutable`'s init function. where the virtual mesh group checks if there is already a physical mesh group. If not, it creates a new one. Otherwise, it checks whether it's compatible with the existing group and throws an error when it's not compatible